dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -239,6 +252,67 @@ def test_correct_duplicate_row_count_test(self):
         self.assertEqual(2999, self.trained_schema.total_samples)
         self.assertEqual(0.0, self.trained_schema._get_duplicate_row_count())
 
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'ColumnDataLabelerCompiler')
+    @mock.patch('dataprofiler.profilers.profile_builder.DataLabeler')
+    def test_correlation(self, *mock):
+        # data with one column, correlation not updated
+        data = pd.DataFrame([1.0, None, 1.0, None, 5.0])
+        profiler = dp.StructuredProfiler(data)
+        self.assertIsNone(profiler.correlation_matrix)
+
+        # data with two columns, but only one is numerical,
+        # then correlation is not updated
+        data = pd.DataFrame([
+            ['test1', 1.0],
+            ['test2', None],
+            ['test1', 1.0],
+            [None, None]])
+        profiler = dp.StructuredProfiler(data)
+        self.assertIsNone(profiler.correlation_matrix)
+
+        # data with multiple numerical columns
+        data = pd.DataFrame({'a': [3, 2, 1, 7, 5, 9, 4, 10, 7, 2],
+                             'b': [10, 11, 1, 4, 2, 5, 6, 3, 9, 8],
+                             'c': [1, 5, 3, 5, 7, 2, 6, 8, 1, 2]})
+        profiler = dp.StructuredProfiler(data)
+        expected_corr_mat = data.corr()
+        pd.util.testing.assert_frame_equal(expected_corr_mat,
+                                           profiler.correlation_matrix)
+
+        # data with multiple numerical columns, with nan values
+        data = pd.DataFrame({'a': [np.nan, np.nan, 1, 7, 5, 9, 4, 10, 7, 2],
+                             'b': [10, 11, np.nan, 4, 2, 5, 6, 3, 9, 8],
+                             'c': [1, 5, 3, 5, 7, 2, 6, 8, np.nan, np.nan]})
+        profiler = dp.StructuredProfiler(data)
+        self.assertIsNone(profiler.correlation_matrix)
+
+        # data with only one numerical columns without nan values
+        data = pd.DataFrame({'a': [np.nan, np.nan, 1, 7, 5, 9, 4, 10, 7, 2],
+                             'b': [10, 11, np.nan, 4, 2, 5, 6, 3, 9, 8],
+                             'c': [1, 5, 3, 5, 7, 2, 6, 8, np.nan, np.nan]})
```

Comment:
all 3 still have nans here, or am i reading it wrong?

Also, maybe add a case where only 1 column has NaN?

Reply:
added